docs/application.md

```diff
@@ -1,105 +1,77 @@
 
-@injectable()
 export class MyService {
 
-    constructor(
-        @inject(Logger)
-        protected logger: Logger
-    ) {}
+    @dep() logger!: Logger;
 
     myMethod() {
         this.logger.info('Hello');
     }
 }
 ```
 
-In this example `MyService` declares `logger` as its dependency. `@inject(Logger)` decorator specifies that this dependency must be resolved by *service identifier* `Logger` (which happens to be an `abstract class`). Inversify also supports String and Symbol service identifiers, but we prefer using constructors.
+In this example `MyService` declares `logger` as its dependency. `@dep()` decorator specifies that this dependency must be resolved by *service identifiers* `Logger`, which is inferred from the property type (in our example `Logger` also happens to be an `abstract class`).
 
-> Think of service identifiers as "keys" of imaginary Map, which associates it with an implementation.
+> Think of service identifiers as keys of imaginary Map, which associates it with an implementation.
 
 Again, `Logger` here acts both as a service identifier and an "interface" which declares logging methods like `.info(...)`. The actual implementation of such methods is unknown to `MyService`, which makes these two components loosely coupled.
 
 > Note: TypeScript `interface` cannot be used as a service identifier, because it is erased by the compiler and thus is not available in runtime. Instead, we use `abstract class` both as a "module interface" (which describes its contract) and as a service identifier to bind implementations.
 
 ## Composition Root
 
-The service identifiers are wired with implementations in a single place called _composition root_. By convention each application should have their composition root defined in `src/main/app.ts`:
+The service identifiers are wired with implementations in a single place called _composition root_.
+By convention each application should have their composition root defined in `src/main/app.ts`:
 
 ```ts
 import { Application } from './framework';
 
 export class App extends Application {
-    constructor() {
+
+    defineGlobalScope(mesh: Mesh) {
         // Bind Logger service identifier to MyLogger class
-        this.container.bind(Logger).to(MyLogger);
+        mesh.service(Logger, MyLogger);
         // Bind MyService service identifier to the same MyService class
-        this.container.bind(MyService).toSelf();
+        mesh.service(MyService);
     }
 }
 ```
 
-When the wiring is done, you can request an instance of any service from application container:
+`Mesh` instance above acts as an IoC container. It maps service identifiers to service constructors.
+
+When the wiring is done, you can request an instance of any service from application mesh:
 
 ```ts
 const app = new App();
-const myService = app.container.get<MyService>(MyService);
-// myService instance is created by Application, and all its dependencies are satisfied
+const myService = app.mesh.resolve(MyService);
```

Comment:
A bit unclear where `app.mesh` comes from because in the code snipped above `mesh` property isn't defined for the `class App`.